### DISCUSSION TITLE
Component testing. Module parse failed. You may need an appropriate loader. React with Webpack.

### DISCUSSION THREAD

**Discussion by Kost927** [cypress-io/cypress#23059] [Component Testing]
Trying to configure cypress component testing into my application. I Have a custom webpack config and got this error.
<img width="507" alt="Screenshot 2022-08-02 at 12 48 24" src="https://user-images.githubusercontent.com/64188262/182345943-f590f386-afb4-4e23-9057-63cbf179d2bc.png">
here is my loader for cypress.
`{`
      `test: /\.ts|\.tsx$/,`
      `exclude: /node_modules/,`
          `use: {`
               `loader: 'babel-loader',`
                   `options: {`
                            `presets: ['@babel/preset-env', '@babel/preset-react'],`
                   `},`
          `},`
  `}`
 PS: [This link ](https://github.com/cypress-io/cypress-component-examples/tree/main/setup-webpack-react-app)from the documentation doesn't work.

**Comment by varunaswal** [comment]
Having the same issue

**Comment by varunaswal** [comment]
Hi, so it seems for me the issue was occurring because I export a function from my `webpack.config.js` - I added `webpackConfig: webpackConfig()` (with relevant function parameters) to my cypress config and the tests are now compiling successfully. However now I'm getting `Timed out waiting for the browser to connect. Retrying again...` - not sure if it's related or a separate issue.

**Comment by Kost927** [comment]
@varunaswal thank you. Also want to try such way
![MicrosoftTeams-image](https://user-images.githubusercontent.com/64188262/183027662-3be42cfc-069f-4d9f-b26f-c08cb8fb483a.png)
Got this answer from discord.
